dragonchain/lib/dto/bnb.py

```diff
@@ -0,0 +1,319 @@
+# Copyright 2019 Dragonchain, Inc.
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#      6. Trademarks. This License does not grant permission to use the trade
+#         names, trademarks, service marks, or product names of the Licensor
+#         and its affiliates, except as required to comply with Section 4(c) of
+#         the License and to reproduce the content of the NOTICE file.
+# You may obtain a copy of the Apache License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+
+import base64
+from typing import Dict, Any
+from ast import literal_eval
+
+from binance_chain.messages import TransferMsg, Signature
+from binance_chain.wallet import Wallet
+from binance_chain.environment import BinanceEnvironment
+import requests
+
+from dragonchain import logger
+from dragonchain import exceptions
+from dragonchain.lib.dto import model
+
+
+DC_MAINNET_NODE = "http://10.2.1.197"  # Mainnet BNB
+DC_TESTNET_NODE = "https://data-seed-pre-0-s3.binance.org:443/"  # Testnet BNB
+# we currently don't have a private testnet node set up...
+
+MAINNET_RPC_PORT = ":27147/"
+MAINNET_API_PORT = ":1169/api/v1/"
+
+CONFIRMATIONS_CONSIDERED_FINAL = 1  # https://docs.binance.org/faq.html#what-is-the-design-principle-of-binance-chain
+BLOCK_THRESHOLD = 3  # The number of blocks that can pass by before trying to send another transaction
+SEND_FEE = 37500  # transfer fee fixed at 0.000375 BNB : https://docs.binance.org/trading-spec.html#current-fees-table-on-mainnet
+
+_log = logger.get_logger()
+
+
+def new_from_user_input(user_input: Dict[str, Any]) -> "BinanceNetwork":
+    """Create a new BinanceNetwork model from user input
+    Args:
+        user_input: User dictionary input (assumed already passing create_binance_interchain_schema)
+    Returns:
+        Instantiated BinanceNetwork client
+    Raises:
+        exceptions.BadRequest: With bad input
+    """
+    dto_version = user_input.get("version")
+    if dto_version == "1":
+        if not user_input.get("private_key"):
+            # We need to create a private key if not provided
+            wallet = Wallet.create_random_wallet()
+            user_input["private_key"] = wallet.private_key
```

Comment:
I'm assuming wallet.private_key from the sdk isn't a base64 encoded private key which is how we want to store it...

check what this is and convert if needed

Reply:
Fixed.